packages/react-native-codegen/src/parsers/parsers-commons.js

```diff
@@ -817,6 +825,36 @@ function propertyNames(
     .filter(Boolean);
 }
 
+function extendsForProp(
+  prop: PropAST,
+  types: TypeDeclarationMap,
+  propName: 'argument' | 'expression',
+): ExtendedPropResult {
+  if (!prop[propName]) {
```

Comment:
The way we are trying to handle these is by using the `Parser` object.

The [`Parser`](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/parser.js) object is an abstract class which is implemented by a [`TypeScriptParser`](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/typescript/parser.js) and a [`FlowParser`](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/flow/parser.js).

So, what we should do is to extends the `Parser` class so that it has a function like:
```ts
argumentForProp(prop: PropAST): $FlowFixMe; // <-- You may have to use $FlowFixMe or define what a PropAST is
```

And implement the function in the two parsers
```ts
// In the FlowParser
argumentForProp(prop: PropAST): $FlowFixMe {
  return prop.argument
}

// In the TypeScriptParser
argumentForProp(prop: PropAST): $FlowFixMe {
  return prop.expression
}
```

Reply:
Oh yes, This is much neater! Thanks for explaining @cipolleschi